packages/core/src/external/mpi/match-patients.ts

```diff
@@ -0,0 +1,171 @@
+import { intersectionWith, isEqual } from "lodash";
+import { PatientDataMPI } from "./patient-incoming-schema";
+import { normalizePatientDataMPI } from "./normalize-patient";
+import jaroWinkler from "jaro-winkler";
+
+// Define a type for the similarity function
+type SimilarityFunction = (
+  patient1: PatientDataMPI,
+  patient2: PatientDataMPI,
+  threshold: number
+) => boolean;
+
+/**
+ * `matchPatients` filters patients based on a similarity function and a threshold.
+ * @param {SimilarityFunction} similarityFunction - Determines if a patient is a match.
+ * @param {Patient[]} patients - Array of patients.
+ * @param {Patient} demo - Patient data to find matches for.
+ * @param {number} threshold - Minimum similarity score for a match.
+ * @param {boolean} greedy - If true, returns the first match. If false, returns all matches.
+ * @returns matched patients.
+ */
+export const matchPatients = (
+  similarityFunction: SimilarityFunction,
+  patients: PatientDataMPI[],
+  demo: PatientDataMPI,
+  threshold: number,
+  greedy = false
+): PatientDataMPI[] => {
+  console.log("patients", patients);
+  console.log("demo", demo);
+  const matchFunction = (patient: PatientDataMPI) => {
+    const PatientDataMPI = normalizePatientDataMPI(patient);
+    if (!PatientDataMPI) {
```

Comment:
as aformentioned, this normalization is redundant if we move towards storing patients data in a normalized way